tests/test_contracts.py

```diff
@@ -125,3 +135,24 @@ def test_scaffolder_extracts_events(scaffolder, test_filesystem):
     new_contract = scaffolder.from_abi(str(path), KNOWN_ADDRESS, "new_contract")
     new_contract.parse_events()
     assert new_contract.events
+
+
+def test_scaffolder_rejects_old_abi(scaffolder, test_filesystem):
+    """Test the scaffolder rejects pre-Solidity 0.6 ABIs."""
+    assert test_filesystem
+    path = Path() / "tests" / "data" / "old_abi.json"
+
+    with pytest.raises(ValueError) as exc_info:
+        scaffolder.from_abi(str(path), KNOWN_ADDRESS, "new_contract")
+
+    error_message = str(exc_info.value)
+    assert "Error processing ABI file:" in error_message
+    assert all(
+        expected in error_message
+        for expected in [
+            "Outdated ABI format detected",
```

Comment:
have an error message constant and then reference it in the test and the error. 
Makes less code duplication for future :)